### PR TITLE
Add gg-deals extension

### PR DESCRIPTION
**Pull request by whitespacecode**
## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by mil3na** [comment]
Hi! Thank you so much for your contribution to Raycast!
My name is Milena and I will be reviewing your extension today :)

About the UI, the only comment I have is that this Small/Medium dropdown is not clear enough about what it does until you actually click it. Maybe make it say `Small Icons` or something similar, instead?
CleanShot 2023-06-27 at 14 21 33@2x

Also, have you considered adding a message on empty states (when nothing is found, for instance) to visit the website instead? Not necessary, but a suggestion. 

Please take a look at the following code comments as well.

File: extensions/gg-deals/src/utils/fetchHtml.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { showToast, Toast } from "@raycast/api";
+import axios from "axios";
+import UserAgent from "user-agents";
+import { baseUrl } from "..";
+
+export async function fetchHtml(url: string, signal?: AbortSignal): Promise<string> {
+  try {
+    const result = await axios.get(`${url}`, {
+      // followRedirect: false,
```
Comment (mil3na) [inline extensions/gg-deals/src/utils/fetchHtml.ts:9]: Remove comment if it's not necessary for documentation

File: extensions/gg-deals/src/utils/types.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    this.linkWidget = linkWidget;
+    this.priceOfficial = priceOfficial;
+    this.priceKeyshops = priceKeyshops;
+    this.pricesByPriceLabels = pricesByPriceLabels;
+    this.pricesByTimeLabels = pricesByTimeLabels;
+    this.editions = editions;
+  }
+}
+
+export const genreColors: { [key: string]: string } = {
```
Comment (mil3na) [inline extensions/gg-deals/src/utils/types.ts:79]: Constants are not types. I suggest you to create a different file for constants.

File: extensions/gg-deals/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,250 @@
+import { useState, useEffect, useCallback } from "react";
+import { ActionPanel, Action, Grid, List, Icon, useNavigation, showToast, Toast } from "@raycast/api";
+import { load, CheerioAPI } from "cheerio";
+import { Details } from "./detail";
+import { DealEntry, SearchState, genreColors } from "./utils/types";
+import { fetchHtml } from "./utils/fetchHtml";
+
+export const baseUrl = "https://gg.deals";
```
Comment (mil3na) [inline extensions/gg-deals/src/index.tsx:8]: consider adding this to the `constants` as well :)

File: extensions/gg-deals/package.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  "author": "Whitespace",
+  "categories": [
+    "Web"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Search Games...",
+      "description": "Template for showing a static grid",
```
Comment (mil3na) [inline extensions/gg-deals/package.json:16]: consider changing the description for something more aligned with your extension

**Comment by whitespacecode** [comment]
Hi @mil3na ,

I have updated my extensions with your suggestions. Thanks for the review!

File: extensions/gg-deals/package.json [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  "icon": "icon.png",
+  "author": "Whitespace",
+  "categories": [
+    "Web"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Game deals",
```
Comment (mil3na) [inline extensions/gg-deals/package.json:15]: Command's title has to be Title Cased. Expected "Game Deals"

**Review by mil3na — APPROVED** [review]
Looks good to me!

**Pull request merged**